### PR TITLE
misc: Added dependabot config file

### PR DESCRIPTION
**Pull request by Harshil2107**
- Added a yaml file to make dependabot target develop instead of stable.

Change-Id: I5b28c06960c5a346b40e2af8f9284b11d9cc07cd

### CODE REVIEW

File: .github/workflows/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    - package-ecosystem: pip
+      directory: /
+      schedule:
+          interval: monthly
+    # Raise pull requests for version updates
+    # to pip against the `develop` branch
+      target-branch: develop
+    # Labels on pull requests for version updates only
+      labels:
+          - pip dependencies
```
Comment (powerjg) [inline .github/workflows/dependabot.yml]: Is this label right? It won't always be `pip` and will often be `npm`.

I think the right label is `misc`

File: .github/workflows/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+---
+version: 2
+updates:
+    - package-ecosystem: pip
```
Comment (powerjg) [inline .github/workflows/dependabot.yml:4]: I think `npm` is the one this has been targeting the most, right?

**Review by powerjg — COMMENTED** [review]
While you're doing this why don't you also update some other options:
- `[assignees](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#assignees)` Put yourself as the assignee
- `[commit-message](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#commit-message)` with the right `prefix`. I.e., `misc:`

File: .github/workflows/dependabot.yml (hunk 2 above)
Comment (Harshil2107) [reply .github/workflows/dependabot.yml:4]: I can add npm but I dont think we use react/ node.js anywhere in gem5. I think so far all dependabot PRs have been python library updates.

File: .github/workflows/dependabot.yml (hunk 1 above)
Comment (Harshil2107) [reply .github/workflows/dependabot.yml]: I will update label to misc

File: .github/workflows/dependabot.yml (hunk 2 above)
Comment (powerjg) [reply .github/workflows/dependabot.yml:4]: My bad. You're right. I thought that the gem5-resources-manager was written with node.

**Review by powerjg — APPROVED** [review]

**Pull request merged**